src/devices/video/imagetek_i4100.cpp

```diff
@@ -670,31 +681,40 @@ void imagetek_i4100_device::window_w(offs_t offset, uint16_t data, uint16_t mem_
 uint16_t imagetek_i4100_device::scroll_r(offs_t offset) { return m_scroll[offset]; }
 void imagetek_i4100_device::scroll_w(offs_t offset, uint16_t data, uint16_t mem_mask) { COMBINE_DATA(&m_scroll[offset]); }
 
-/****************************************************
+/*
  *
  * Screen Control Register:
  *
  * f--- ---- ---- ----     ?
  * -edc b--- ---- ----
- * ---- -a98 ---- ----     ? Leds (see gakusai attract)
+ * ---- -a98 ---- ----     external control pins
+ *                         \- gakusai attract, unknown purpose
+ *                            (bit 2 enabled during title photo flashes,
+ *                             bit 1 always?,
+ *                             bit 0 periodically during gal sequences);
+ *                         \- (bit 0) karatour/ladykill/3kokushi external irq level 5 enable, (bit 1-2) unknown;
+ *                         \- puzzli [hblank] timer stop/start? 0 during transitions, 7 otherwise
+ *                            (including individual printouts of ROM statuses during POST);
  * ---- ---- 765- ----     16x16 Tiles  (Layer 2-1-0)
  * ---- ---- ---4 32--
  * ---- ---- ---- --1-     Blank Screen
  * ---- ---- ---- ---0     Flip  Screen
  *
- ****************************************************/
+ */
 void imagetek_i4100_device::screen_ctrl_w(offs_t offset, uint16_t data, uint16_t mem_mask)
 {
-	m_layer_tile_select[2] = BIT(data,7);
-	m_layer_tile_select[1] = BIT(data,6);
-	m_layer_tile_select[0] = BIT(data,5);
+	m_ext_ctrl_2_cb(bool(BIT(data, 10)));
+	m_ext_ctrl_1_cb(bool(BIT(data, 9)));
+	m_ext_ctrl_0_cb(bool(BIT(data, 8)));
```

Comment:
Read/write line members use `int`, not `bool` – the cast to `bool` means there’s an additional implicit cast from `bool` to `int`.